domain/secret/service/watcher.go

```diff
@@ -51,44 +51,87 @@ func NewWatchableService(
 // WatchConsumedSecretsChanges watches secrets consumed by the specified unit
 // and returns a watcher which notifies of secret URIs that have had a new revision added.
 func (s *WatchableService) WatchConsumedSecretsChanges(ctx context.Context, unitName coreunit.Name) (watcher.StringsWatcher, error) {
-	ctx, span := trace.Start(ctx, trace.NameFromFunc())
+	_, span := trace.Start(ctx, trace.NameFromFunc())
 	defer span.End()
 
 	tableLocal, queryLocal := s.secretState.InitialWatchStatementForConsumedSecretsChange(unitName)
-	wLocal, err := s.watcherFactory.NewNamespaceWatcher(
-		// We are only interested in CREATE changes because
-		// the secret_revision.revision is immutable anyway.
-		queryLocal,
-		eventsource.NamespaceFilter(tableLocal, changestream.Changed),
-	)
-	if err != nil {
-		return nil, errors.Capture(err)
-	}
-	processLocalChanges := func(ctx context.Context, revisionUUIDs ...string) ([]string, error) {
-		return s.secretState.GetConsumedSecretURIsWithChanges(ctx, unitName, revisionUUIDs...)
-	}
-	sWLocal, err := newSecretStringWatcher(wLocal, s.logger, processLocalChanges)
-	if err != nil {
-		return nil, errors.Capture(err)
+	tableRemote, queryRemote := s.secretState.InitialWatchStatementForConsumedRemoteSecretsChange(unitName)
+
+	initialQuery := func(ctx context.Context, runner coredatabase.TxnRunner) ([]string, error) {
+		localConsumedRevisionIDs, err := queryLocal(ctx, runner)
```

Comment:
Ideally, we can make queryLocal to return the secretURIs instead of having a 2nd state call.